src/preprocessor.js

```diff
@@ -0,0 +1,49 @@
+import { templateResource } from "./templates";
+import { expressResource } from "./expressions";
+
+/**
+ * @param {Array<Objects>} persona 
+ * @returns {Array<Objects} the parsed persona reiterated by the chain of callbacks
+ */
+export const parsePersona = persona => iteratePersona(iteratePersona(iteratePersona(persona, templateResource), expressResource), pathResource);
+
+/**
+ * @param {Array<Objects>} persona entire persona tree
+ * @param {Function} callback custom callback
+ * @returns {Array<Objects} the iterated persona by the callback provided
+ * Walks the persona tree and uses a provided callback function to apply necessary logic to prepare for the seeding
+ */
+function iteratePersona (persona, callback) {
```

Comment:
Rename this to `recurse`, `traverse`, or `recursePersona`, it's doing more than iterating.